### ISSUE TITLE
[Enhancement] List view - when no files are selected, have the 'Home' key highlight the top-listed file

### ISSUE THREAD

**Issue by NintendoManiac64** [linuxmint/nemo#3036]
This is basically a re-creation (albeit with more detail) of what I thought to have been a bug in Mint 21 beta but, according to clefebvre, this is not a bug and is actually a feature request.

Original issue:
- https://github.com/linuxmint/mint21-beta/issues/114

**Describe the bug**
If you have your folder view set to "List View" and you view a folder that has multiple files and no files are highlighted, pressing the "Home" key does not highlight the top-listed file.

**To Reproduce**
1. make sure Nemo's folder view is set to "List View" and that it's also using that by default
2. in Nemo, navigate to a folder with multiple files, such as `/etc/`
3. press the "Home" key on your keyboard
4. observe how no files become highlighted
5. press the back button in Nemo
6. go into the `/etc/` folder again
7. make sure that Nemo is still in "list view"
8. press the "End" key on your keyboard
9. observe how it _does_ highlight the last file that is listed
10. change Nemo to "compact view"
11. press the back button in Nemo
12. go into the `/etc/` again
13. make sure that Nemo is still set to "compact view"
14. press the "Home" key on your keyboard
15. observe how the first-listed file is now highlighted
16. press the back button in Nemo
17. go into the `/etc/` folder again
18. make sure that Nemo is still in "compact view"
19. press the "End" key on your keyboard
20. observe how the last-listed file is now highlighted
21. change Nemo to "icon view"
22. press the back button in Nemo
23. go into the `/etc/` again
24. make sure that Nemo is still set to "icon view"
25. press the "Home" key on your keyboard
26. observe how the first-listed file is now highlighted
27. press the back button in Nemo
28. go into the `/etc/` folder again
29. make sure that Nemo is still in "compact view"
30. press the "End" key on your keyboard
31. observe how the last-list file is now highlighted

**Expected behavior**
Simply put, I would think that it would make sense that pressing the "Home" key on your keyboard would highlight the first-listed file when using "list view".

**Comment by Jeremy7701** [comment]
If you press the END key followed by the HOME key in LMDE5 this will select the first item.
Maybe the same workaround will work in Mint?

**Comment by NintendoManiac64** [comment]
> If you press the END key followed by the HOME key in LMDE5 this will select the first item.
Maybe the same workaround will work in Mint?

The 'Home' key already works if there's a file that's already highlighted which is what is occurring by pressing 'End' and then pressing 'Home'

**Comment by NintendoManiac64** [comment]
I can confirm this still exactly occurs as-is in Nemo v6.6.3 (which is also the version that comes bundled with Mint 22.3 Cinnamon).

Now being 4 years wiser in terms of Linux wisdom, I'm now aware this is actually a GTK bug as it impacts not just Nemo [but also Thunar](https://gitlab.xfce.org/xfce/thunar/-/issues/844) and PCManFM. It's also possible that step 39 on issue #3032 is basically a recreation of this bug but with different keyboard keys since it's the *only* step of issue #3032 that also occurs in Thunar to this day.

So I'm not really sure of the state of keeping this open or closed...
